src/contexts/ThemeContext.jsx

```diff
@@ -5,7 +5,13 @@ import { Theme } from '../ui/themes';
 const ThemeContext = createContext();
 
 const ThemeContextProvider = ({ children }) => {
-  const [theme, setTheme] = useState(Theme.light);
+  const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+  const systemTheme = mediaQuery.matches ? Theme.dark : Theme.light;
+  const [theme, setTheme] = useState(systemTheme);
+
+  mediaQuery.addEventListener('change', (event) => {
```

Comment:
We add an event listener to the mediaQueries browser global -- when the user system changes, this will fire the callback and set the theme to the current user's preferences.